### ISSUE TITLE
Make mid2agb output match

### ISSUE THREAD

**Issue by yenatch** [pret/pokeruby#311]
Right now all the songs in `sound/songs/` have to be `.s` to match, since mid2agb can't convert to midi and back and still match. When this is fixed, the songs can be converted to midi.

**Comment by ProjectRevoTPP** [comment]
IIRC YamaArashi was having issues with the midi compression used with matching the `.s` files.

**Comment by Keitaro37** [comment]
a while ago I started attempting to recreate the MIDI files byte by byte so that when compiled they would match the .s files exactly. This...proved to be stupidly tedious. still, the commands are very detailed within the M4a engine's files, which was helpful. Perhaps creating some type of script to generate a midi with the exact commands in the exact order as the .s file would do the trick, as that's exactly what I was doing by hand and it matched up perfectly.

**Comment by Keitaro37** [comment]
(actually, I seem to recall there was a file in the m4a engine documentation which further elaborated on various MIDI commands, if it would help, I can dig it up)

**Comment by yenatch** [comment]
Yeah. Yama had a tool for converting .s to midi but it was lost, so we will have to recreate it.

**Comment by camthesaxman** [comment]
I have a modified version of GBA Mus Riper, which was able to produce a matching bgm_pokecen.mid. https://github.com/camthesaxman/pokeruby_mus_riper I haven't tried any other songs yet. We still need to fix our mid2agb so that it can handle all of the MIDI commands.
